pkg/store/consul/statusstore/rcstatus/store.go

```diff
@@ -25,41 +27,46 @@ func NewConsul(statusStore statusstore.Store, namespace statusstore.Namespace) C
 	}
 }
 
-func (c ConsulStore) Get(rcID fields.ID) (RCStatus, *api.QueryMeta, error) {
+func (c ConsulStore) Get(rcID fields.ID) (Status, *api.QueryMeta, error) {
 	if rcID == "" {
-		return RCStatus{}, nil, util.Errorf("Provided replication controller ID was empty")
+		return Status{}, nil, util.Errorf("Provided replication controller ID was empty")
 	}
 
-	status, queryMeta, err := c.statusStore.GetStatus(statusstore.RC, statusstore.ResourceID(rcID), c.namespace)
+	rawStatus, queryMeta, err := c.statusStore.GetStatus(statusstore.RC, statusstore.ResourceID(rcID), c.namespace)
 	if err != nil {
-		return RCStatus{}, queryMeta, err
+		return Status{}, queryMeta, err
 	}
 
-	rcStatus, err := statusToRCStatus(status)
+	status, err := rawStatusToStatus(rawStatus)
 	if err != nil {
-		return RCStatus{}, queryMeta, err
+		return Status{}, queryMeta, err
 	}
 
-	return rcStatus, queryMeta, nil
+	return status, queryMeta, nil
 }
 
-func (c ConsulStore) Set(rcID fields.ID, status RCStatus) error {
+func (c ConsulStore) Set(rcID fields.ID, status Status) error {
 	if rcID == "" {
 		return util.Errorf("Provided replication controller ID was empty")
 	}
 
-	rawStatus, err := rcStatusToStatus(status)
+	rawStatus, err := statusToRawStatus(status)
 	if err != nil {
 		return err
 	}
 
 	return c.statusStore.SetStatus(statusstore.RC, statusstore.ResourceID(rcID), c.namespace, rawStatus)
 }
 
-func (c ConsulStore) Delete(rcID fields.ID) error {
+func (c ConsulStore) CASTxn(ctx context.Context, rcID fields.ID, modifyIndex uint64, status Status) error {
```

Comment:
might be worth a test making sure that this function mutates a status when you pass the correct index, and fails to when you pass the wrong one

Reply:
```
func (s *FakeStatusStore) CASStatus(
	ctx context.Context,
	t statusstore.ResourceType,
	id statusstore.ResourceID,
	namespace statusstore.Namespace,
	status statusstore.Status,
	modifyIndex uint64,
) error {
	return errors.New("CASStatus uses transactions which requires a real consul instance")
}
``` 
Looks like to test `CASTxn()`  I'll need to create a new fake then

Reply:
Done.